data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java

```diff
@@ -34,28 +37,38 @@
-            ServiceMapTestData.DATA_500, ServiceMapTestData.DATA_600, ServiceMapTestData.DATA_700, ServiceMapTestData.DATA_800,
-            ServiceMapTestData.DATA_900, ServiceMapTestData.DATA_1000, ServiceMapTestData.DATA_1100);
-    private static final List<ServiceMapTestData> testDataSet2 = Arrays.asList(
-            ServiceMapTestData.DATA_101, ServiceMapTestData.DATA_201, ServiceMapTestData.DATA_301, ServiceMapTestData.DATA_401,
-            ServiceMapTestData.DATA_501);
+    private static final String TEST_TRACEID_1 = "ABC";
+    private static final String TEST_TRACEID_2 = "CBA";
+    private static final int DATA_PREPPER_PORT_1 = 21890;
+    private static final int DATA_PREPPER_PORT_2 = 21891;
+    private static final List<ServiceMapTestData> testDataSet11 = Arrays.asList(
+            ServiceMapTestData.DATA_100, ServiceMapTestData.DATA_200, ServiceMapTestData.DATA_500, ServiceMapTestData.DATA_600,
+            ServiceMapTestData.DATA_700, ServiceMapTestData.DATA_1000);
+    private static final List<ServiceMapTestData> testDataSet12 = Arrays.asList(
+            ServiceMapTestData.DATA_300, ServiceMapTestData.DATA_400, ServiceMapTestData.DATA_800,
+            ServiceMapTestData.DATA_900, ServiceMapTestData.DATA_1100);
+    private static final List<ServiceMapTestData> testDataSet21 = Arrays.asList(
+            ServiceMapTestData.DATA_101, ServiceMapTestData.DATA_201, ServiceMapTestData.DATA_401, ServiceMapTestData.DATA_501);
+    private static final List<ServiceMapTestData> testDataSet22 = Collections.singletonList(ServiceMapTestData.DATA_301);
     private static final String SERVICE_MAP_INDEX_NAME = "otel-v1-apm-service-map";
 
     @Test
     public void testPipelineEndToEnd() throws IOException, InterruptedException {
         // Send test trace group 1
-        final ExportTraceServiceRequest exportTraceServiceRequest1 = getExportTraceServiceRequest(
-                getResourceSpansBatch("ABC", testDataSet1)
+        final ExportTraceServiceRequest exportTraceServiceRequest11 = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_TRACEID_1, testDataSet11)
+        );
+        final ExportTraceServiceRequest exportTraceServiceRequest12 = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_TRACEID_1, testDataSet12)
         );
 
-        sendExportTraceServiceRequestToSource(exportTraceServiceRequest1);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_1, exportTraceServiceRequest11);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_2, exportTraceServiceRequest12);
 
         //Verify data in elasticsearch sink
+        final List<ServiceMapTestData> testDataSet1 = Stream.of(testDataSet11, testDataSet12)
+                .flatMap(Collection::stream).collect(Collectors.toList());
+        final List<Map<String, Object>> possibleEdges = getPossibleEdges(TEST_TRACEID_1, testDataSet1);
```

Comment:
I think this is the proper way of implicitly verifying the peer forwarding (for service map), good job.

Reply:
Credits to @kowshikn